src/Components/test/testassets/TestServer/TestCircuitContextAccessor.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.Server.Circuits;
+
+namespace TestServer;
+
+public class TestCircuitContextAccessor : CircuitHandler, ICircuitInboundEventHandler
+{
+    private readonly AsyncLocal<bool> _hasCircuitContext = new();
+
+    public bool HasCircuitContext => _hasCircuitContext.Value;
+
+    public async Task HandleInboundEventAsync(CircuitInboundEventContext context, CircuitInboundEventDelegate next)
+    {
+        _hasCircuitContext.Value = true;
+        await next(context);
+        _hasCircuitContext.Value = false;
+    }
+}
```

Comment:
It will be interesting that we update the docs for HttpClientFactory with this approach, as well as offer other examples, like making values available to OwningComponentBase and other components.